ui/app/appointments/constants.js

```diff
@@ -22,6 +22,8 @@ Bahmni.Appointments.Constants = (function () {
         defaultCalendarSlotDuration: "00:30",
         defaultCalendarStartTime: "08:00",
         defaultCalendarEndTime: "19:00",
+        defaultExpandServiceFilter: -1,
```

Comment:
If we do not have these constants then 
                $scope.expandToDepth = appService.getAppDescriptor().getConfigValue('collapseServiceFilter') == true ?
                    Bahmni.Appointments.Constants.collapseServiceFilter : Bahmni.Appointments.Constants.defaultExpandServiceFilter; line would look like
                $scope.expandToDepth = appService.getAppDescriptor().getConfigValue('collapseServiceFilter') == true ?
                    0 : -1;
In order to avoid 0 and 1 to be used directly as random numbers, we introduced the two constants. 
Also if the configuration is not mentioned, the default expanded view would be rendered to the user.

Reply:
Wouldn't it still be better than someone having to define 2 constants, which seem redundant? Also, if its condition behavior, please use booleans than numeric 0, -1, 1 (unless such numerics are used as codes, which does not seem to be case here).

Reply:
Agreed. But we need 0, -1 specifically as ivh-treeview-expand-to-depth="-1" represents expanding the leaf nodes and 0 represents collapsing the same. Can we still represent it as boolean?

Reply:
I think so. its just a flag to check whether a node is expanded or collapsed right? then boolean makes sense.